pandas/core/groupby/groupby.py

```diff
@@ -1275,8 +1275,16 @@ def std(self, ddof: int = 1):
         Series or DataFrame
             Standard deviation of values within each group.
         """
-        # TODO: implement at Cython level?
-        return np.sqrt(self.var(ddof=ddof))
+        result = self.var(ddof=ddof)
+        if result.ndim == 1:
+            result = np.sqrt(result)
+        else:
+            cols = result.columns.get_indexer_for(
+                result.columns.difference(self.exclusions).unique()
+            )
+            result.iloc[:, cols] = np.sqrt(result.iloc[:, cols]).values
```

Comment:
is there a reason you are using .values here?

Reply:
Without values, iLocIndexer._setitem_with_indexer takes a broken path when the index labels are not unique. It goes from using positional indexes to labels because of the BlockManager, which causes errors when the labels are not unique. This causes tests in test_groupby to fail. By using values, we are able to get around it.

Reply:
can you reproduce this bug independently of this PR? please open a separate issue, this is a problem that needs fixing.

Reply:
.iloc should not be aligning at all